### PR TITLE
Basic analysis

### PR DESCRIPTION
**Pull request by rnbndr**
Basic analysis of the dataset containing information about:
- number of rows and columns
- missing values per column
- rows with a lot of missing values
- outlier detection

### CODE REVIEW

**Comment by bican-guel** [comment]
Sorry for the possible millions of messages for my reviews and comments. I was having issues with Github.

File: controller/managers/DataSetAnalysisManager.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+            current_column_sorted = current_column.sort_values()
+
+            Q1, Q2, Q3 = np.percentile(current_column_sorted, [25,50,75])
+
+            if ((Q3 - Q1) < 0.1) :
+                continue
+            
+            # interquartile range
+            IQR = Q3 - Q1
```
Comment (jnnks) [inline controller/managers/DataSetAnalysisManager.py:120]: Why not name IQR `interquartile_range`? :D

**Review by jnnks — CHANGES_REQUESTED** [review]
You are passing around and returning the jsonfile quite a lot.
I think you should change your functions to only take the dataset as an input and then return any output as a separate value. Then the analysis dictionary can be built together in the startAnalysis function:
```python
 jsonfile["basic analysis"] = {
  "outlier": __detect_outliers(dataset),
  ...
}
```
That way we can use the functions in the module without having to construct a result dictionary and still get meaningful results. It also reduces the amount of hardcoded string keys.
____
**Optional**
Think about the names you give your keys and variables. They should be short, but unique and descriptive.
Without reading the right side of an assignment what is the difference between `column` and `current_column` in line 108+? What about `current_column` in line 126? :D

Please remove the spaces in dictionary keys. Treat them just like variable names.
You will not remember whether the key is `rows with many missing values` or `rows with missing values` and neither will someone else :) use something like `high_na_rows`

____

The logic seems good to me. I'm sure some statements can be shorter like line 79+, but that's not important as long as it works and can be understood.
I'm happy so far, if you add the comments and restructure the methods

File: controller/managers/DataSetAnalysisManager.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        Return the JSON file with additional data
+        """
+
+        jsonfile["basic analysis"]["missing values in columns"] = {}
+        jsonfile["basic analysis"]["rows with many missing values"] = {}
+
+        dataset_missings_columns = dataset.isna().sum()
+        for index, value in dataset_missings_columns.items():
+
+            jsonfile["basic analysis"]["missing values in columns"][index] = value
```
Comment (jnnks) [inline controller/managers/DataSetAnalysisManager.py:81]: Can be shorter:
```python
# map all columns with the number of missing values
na_counts = dict(dataset.isna().sum().items())
```

**Comment by bican-guel** [comment]
@jnnks 
Just a friendly reminder.
All your requests have been fulfilled.

As its your initial review please approve for the merge.^^

**Comment by jnnks** [comment]
@rnbndr could you fix the methods return types?
I assume the last two methods `__detect_duplicate_columns` and `__detect_duplicate_rows` are a work in progress?
They do not seem to make a change. Could you take them out of this PR in this case? We should not merge WIP code.

**Review by jnnks — APPROVED** [review]

**Comment by jnnks** [comment]
squashed and merged
I'll leave the branch so you can continue working on it :)

**Pull request merged**